Source/AFResult.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  AFResult.swift
+//
+//  Copyright (c) 2014-2019 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+public typealias AFResult<T> = Result<T, Error>
+
+
+// MARK: AFExtension
+
+extension Result {
+    /// Type that acts as a extension point for all `Result` types.
+    public struct AFExtension {
+        let result: Result<Success, Failure>
+    }
+
+    public static var af: AFExtension.Type { return AFExtension.self }
+    public var af: AFExtension { return AFExtension(result: self) }
+}
+
+// MARK: - Public APIs
+
+extension Result.AFExtension {
```

Comment:
The public / internal distinction here seems arbitrary. Why don't we keep anything we want to make public namespaced, and any other extensions we need should be internal and non-namespaced? So we could just mark this extension public and move anything internal to the `AFResult` extension.

Reply:
If the internal extensions are only needed to support unit tests, moving those additions to the test target might be a good idea.

Reply:
That's exactly what I did @jshier. I deleted any of them we don't actually need, kept the ones public in the namespaced extension, and moved the initializer to be an internal API on `AFResult`. Are you referring to what I did in the test suite by adding internal API behind the namespace? I could certainly move those to just extend `AFResult`.

Reply:
And @AtomicCat, the initializer at the bottom of the `AFResult` file is needed in the framework as well as the test target.

Reply:
My point was that any extension that isn't public doesn't need to be namespaced and so can be moved directly into the `AFResult` extension four our internal use. If we want to offer any public extensions, we should discuss which features those would be. I think that extensions here could all be public, and so the extension should be made public. I've seen `isSuccess` / `isFailure` misused quite a bit in the wild, so keeping those internal would be good.

Reply:
We decided to remove the public extensions altogether in Slack.